Frameworks/CoreGraphics/CGGradient.mm

```diff
@@ -150,13 +149,18 @@ CGGradientRef CGGradientCreateWithColorComponents(CGColorSpaceRef colorSpace,
 
 /**
  @Status Interoperable
+ @Note Only RGB colorspace is supported, but individual colors can be of different colorspace
 */
 CGGradientRef CGGradientCreateWithColors(CGColorSpaceRef colorSpace, CFArrayRef colors, const CGFloat* locations) {
-    RETURN_NULL_IF(!colorSpace);
     RETURN_NULL_IF(!colors);
     RETURN_NULL_IF(!__isValidGradientColorspaceModel(colorSpace));
     // location can be null
-    return __CGGradient::CreateInstance(kCFAllocatorDefault, colorSpace, colors, locations);
+    static woc::StrongCF<CGColorSpaceRef> colorSpaceRGB;
+    if (!colorSpace) {
```

Comment:
this is using a static in a non-threadsafe way; also, it's moderately confusing as to what's happening